file/index.js

```diff
@@ -1,120 +1,120 @@
 'use strict';
 
 const AWS = require('aws-sdk');
-var s3 = new AWS.S3();
+let s3 = new AWS.S3();
 
-var bucketName = 'lambda-function-records';
-var bucketParams = { Bucket: bucketName };
+let bucketName = 'lambda-function-records';
+let bucketParams = {
+  Bucket: bucketName
+};
 
-var limitNumberOfFileToUpdateInOneRun = 2;
+let limitNumberOfFileToUpdateInOneRun = 2;
 
 exports.handler = (request, context, callback) => {
-    update(request.state, request.secrets, callback);
+  update(request.state, request.secrets, callback);
 };
 
-async function update(state, secrets, callback) {
-    let response = initializeResponse(state);
-
-    let modifiedFiles = [];
-
-    let listObjectsPromise = s3.listObjects(bucketParams).promise();
-
-    listObjectsPromise.then(function (result) {
-        for (var index = 0; index < result.Contents.length; index++) {
-            
-            // Only JSON files to be processed
-            // We can add some pattern
-            if (!result.Contents[index].Key.endsWith(".tsv") || Date.parse(result.Contents[index].LastModified) <= Date.parse(state.since)) {
-                continue;
-            }
-
-            modifiedFiles.push(result.Contents[index]);
-            
-            // If we want to process limited number of files in one lambda execution, so we should only add those number of files
-            if (modifiedFiles.length === limitNumberOfFileToUpdateInOneRun) {
-                response.hasMore = true;
-                break;
-            }
+async function update(state, secrets, onError) {
+  let response = initializeResponse(state);
+
+  let modifiedFiles = [];
+
+  let listObjectsPromise = s3.listObjects(bucketParams).promise();
+
+  listObjectsPromise.then((result) => {
+    for (let index = 0; index < result.Contents.length; index++) {
+
+      // Only TSV files to be processed
+      // We can add some pattern
+      if (!result.Contents[index].Key.endsWith(".tsv") || Date.parse(result.Contents[index].LastModified) <= Date.parse(state.since)) {
+        continue;
+      }
+
+      modifiedFiles.push(result.Contents[index]);
+
+      // If we want to process limited number of files in one lambda execution, so we should only add those number of files
+      if (modifiedFiles.length === limitNumberOfFileToUpdateInOneRun) {
+        response.hasMore = true;
+        break;
+      }
+    }
+  }).catch((err) => {
+    onError(err); // Return when some error occurred while listing objects in bucket
+    console.log("Error in listing bucket", JSON.stringify(err) + "\n");
+  });
+
+  // Waiting for listing of objects so we can get all modified files
+  await listObjectsPromise;
+
+  // Sort in ascending order
+  modifiedFiles.sort((a, b) => {
+    return a - b;
```

Comment:
wait, we are comparing `LastModified`, not objects?!

Reply:
I changed code to sort based on `LastModified` https://github.com/fivetran/functions/pull/1/commits/0596a91870f5e84ca56b1c7a3791be49260477a9